packages/astro/src/check.ts

```diff
@@ -0,0 +1,115 @@
+import { AstroCheck, DiagnosticSeverity } from '@astrojs/language-server';
+import type { AstroConfig } from './@types/astro';
+import { bold, blue, black, bgWhite, red, cyan, yellow } from 'kleur/colors';
+import glob from 'fast-glob';
```

Comment:
@natemoo-re not sure if you resolved that "fdir isn't sending back good results" issue in `Astro.fetchContent()`, but using fast-glob could be a good alternative.